acnportal/acnsim/network/charging_network.py

```diff
@@ -163,22 +228,88 @@ def update_pilots(self, pilots, i, period):
+        ids = self.station_ids
+        for station_number in range(len(ids)):
+            new_rate = pilots[station_number, i]
+            self._EVSEs[ids[station_number]].set_pilot(new_rate, self._voltages[station_number], period)
+
+    def constraint_current(self, schedule_matrix, constraints=None, time_indices=None, linear=False):
+        """ Return the aggregate currents subject to the given constraints. If constraints=None,
+        return all aggregate currents.
+
+        Args:
+            schedule_matrix (np.Array): 2-D matrix with each row corresponding to an EVSE and each
+                column corresponding to a time index in the schedule.
+            constraints (List[str]): List of constraint id's for which to calculate aggregate current. If
+                None, calculates aggregate currents for all constraints.
+            time_indices (List[int]): List of time indices for which to calculate aggregate current. If None, 
+                calculates aggregate currents for all timesteps.
+            linear (bool): If True, linearize all constraints to a more conservative but easier to compute constraint by
+                ignoring the phase angle and taking the absolute value of all load coefficients. Default False.
+
+        Returns:
+            List[complex]: Aggregate currents subject to the given constraints.
+        """
+        # Convert list of constraint id's to list of indices in constraint matrix
+        if constraints is not None:
+            constraint_indices = [i for i in range(len(self.constraint_index)) if self.constraint_index[i] in constraints]
+        else:
+            constraint_indices = list(range(len(self.constraint_index)))
+
+        # If we only want the constraint currents at specific time indices,
+        # index schedule_matrix columns using these indices
+        if time_indices is not None:
+            schedule_matrix = schedule_matrix[:, time_indices]
+
+        if linear:
+            return complex(np.abs(self.constraint_matrix[constraint_indices]@schedule_matrix))
+        else:
+            # build vector of phase angles on EVSE
+            angle_coeffs = np.exp(1j*np.deg2rad(self._phase_angles))
+
+            # multiply schedule by angles matrix element-wise
+            shifted_schedule = (schedule_matrix.T * angle_coeffs).T
+
+            # multiply constraint matrix by current schedule, shifted by the phases
+            return self.constraint_matrix[constraint_indices]@shifted_schedule
+
+    def is_feasible(self, schedule_matrix, t=0, linear=False):
+        """ Return if a set of current magnitudes for each load are feasible.
+
+        Args:
+            schedule_matrix (np.Array): 2-D matrix with each row corresponding to an EVSE and each
+                column corresponding to a time index in the schedule.
+            t (int): Index into the charging rate schedule where feasibility should be checked.
```

Comment:
Lets remove t, the user can slice what they pass to is_feasible if they don't want to check the whole matrix.